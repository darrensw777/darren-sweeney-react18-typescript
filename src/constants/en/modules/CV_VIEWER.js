export const CV_VIEWER = {
    DOWNLOAD: "Click here to download a PDF of my CV",
    CONTACT: "Contact me",
    PAGE: "Page",
    OF: "of",
    NEXT: "Next",
    PREVIOUS: "Previous",
};
