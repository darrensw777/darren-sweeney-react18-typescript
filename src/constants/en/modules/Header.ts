export const HeaderCopy = {
    headerLinks: [
        {
            linkTo: '/',
            label: 'Home',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
        {
            linkTo: '/about',
            label: 'About',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/work',
            label: 'Work',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/music',
            label: 'Music',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/videos',
            label: 'Videos',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/gallery',
            label: 'Gallery',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: false,
        },
        {
            linkTo: '/contact',
            label: 'Contact',
            active: true,
            sideNav: true,
            headerNav: true,
            sideNavFullLine: true,
        },
    ],
};

export default HeaderCopy;
