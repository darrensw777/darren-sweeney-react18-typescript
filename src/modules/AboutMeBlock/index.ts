/* istanbul ignore file */
export { default } from './AboutMeBlock';
