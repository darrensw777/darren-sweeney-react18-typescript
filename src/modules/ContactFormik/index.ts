/* istanbul ignore file */
export { default } from './ContactFormik';
