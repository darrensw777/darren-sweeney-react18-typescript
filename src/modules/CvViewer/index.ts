/* istanbul ignore file */
export { default } from './CvViewer';
