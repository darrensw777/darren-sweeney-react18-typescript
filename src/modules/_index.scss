@import './Header/Header.scss';
@import './Footer/Footer.scss';
@import './PageHeader/page-header.scss';
@import './AboutMeBlock/about-me-block.scss';
@import './CvViewer/cv-viewer.scss';
@import './SoundCloudPlayer/soundcloud-player.scss';
@import './ImagesGallery/images-gallery.scss';
@import './ContactFormik/contact-formik.scss';
@import './InterestLists/interest-lists.scss';