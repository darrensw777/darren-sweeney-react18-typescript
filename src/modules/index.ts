/* istanbul ignore file */
/*
 * Exporting all the components from the components folder,
 * components can be imported as the example below:
 * import { Logo, Avatar } from 'components';
 */

export { default as Header } from './Header';
export { default as Footer } from './Footer';
export { default as PageHeader } from './PageHeader';
export { default as AboutMeBlock } from './AboutMeBlock';
export { default as CvViewer } from './CvViewer';
export { default as SoundCloudPlayer } from './SoundCloudPlayer';
export { default as ImagesGallery } from './ImagesGallery';
export { default as ContactFormik } from './ContactFormik';
export { default as InterestLists } from './InterestLists';
export { default as HomePageBlockContainer } from './HomePageBlockContainer';
