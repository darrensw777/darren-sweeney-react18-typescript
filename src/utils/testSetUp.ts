import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';

const testSetUp = ({ Component }: { Component: ReactElement }) => {
    const { container, getByText, getByRole, asFragment, getAllByRole, getByTestId, getByTitle } = render(Component, {
        wrapper: MemoryRouter,
    });
    return {
        container,
        getByText,
        getByRole,
        asFragment,
        getAllByRole,
        getByTestId,
        getByTitle,
    };
};

export default testSetUp;
