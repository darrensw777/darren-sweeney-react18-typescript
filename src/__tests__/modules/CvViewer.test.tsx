import { fireEvent, waitFor } from '@testing-library/react';
import { CvViewer } from 'modules';
import testSetUp from 'utils/testSetUp';

describe('CvViewer container', () => {
    let getByText: any, asFragment: any, container: any, getByRole: any, getByTestId: any;
    beforeEach(() => {
        ({ asFragment, getByText, container, getByRole, getByTestId } = testSetUp({
            Component: <CvViewer />,
        }));
    });

    test('matches snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the block', () => {
        const blocks = container.getElementsByClassName('cv-viewer');
        expect(blocks.length).toBe(1);
    });

    test('renders the increment and decrement buttons', () => {
        const decrementButton = getByTestId('previousPage');
        const incrementButton = getByTestId('nextPage');
        expect(decrementButton).toBeInTheDocument();
        expect(incrementButton).toBeInTheDocument();
        expect(getByText('Page 1 of 4')).toBeInTheDocument();
    });

    test('checks the text changes re page currently on', () => {
        const decrementButton = getByTestId('previousPage');
        const incrementButton = getByTestId('nextPage');
        fireEvent.click(incrementButton);
        expect(getByText('Page 2 of 4')).toBeInTheDocument();
        fireEvent.click(incrementButton);
        expect(getByText('Page 3 of 4')).toBeInTheDocument();
        fireEvent.click(decrementButton);
        expect(getByText('Page 2 of 4')).toBeInTheDocument();
    });

    test('renders the correct image path', async () => {
        const image = getByRole('img') as HTMLImageElement;
        await waitFor(() => expect(image.src === '/img/cv/ds1.webp'));
    });

    test('renders the correct image path after click', async () => {
        const decrementButton = getByTestId('previousPage');
        const incrementButton = getByTestId('nextPage');
        const image = getByRole('img') as HTMLImageElement;

        await waitFor(() => expect(image.src === '/img/cv/ds1.webp'));
        
        fireEvent.click(incrementButton);
        await waitFor(() => expect(image.src === '/img/cv/ds2.webp'));

        fireEvent.click(incrementButton);
        await waitFor(() => expect(image.src === '/img/cv/ds3.webp'));

        fireEvent.click(decrementButton);
        await waitFor(() => expect(image.src === '/img/cv/ds2.webp'));

        fireEvent.click(decrementButton);
        await waitFor(() => expect(image.src === '/img/cv/ds1.webp'));
    });
});
