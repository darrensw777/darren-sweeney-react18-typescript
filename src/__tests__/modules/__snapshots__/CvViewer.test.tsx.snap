// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvViewer container matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="moduleStyles__CvViewerWrapper-sc-187oehp-0 hHNrHN"
  >
    <div
      class="cv-viewer"
    >
      <div
        class="links"
      >
        <a
          class="download-pdf"
          href="/pdf/DarrenSweeney5.0.pdf"
          target="_blank"
        >
          Click here to download a PDF of my CV
        </a>
        <a
          href="/contact"
        >
          Contact me
        </a>
      </div>
      <div
        class="cv-page-container"
      >
        <img
          alt="CV page number 2"
          src="/img/cv/ds1.webp"
        />
      </div>
      <div
        class="componentStyles__CvPaginationWrapper-sc-1wsblha-0 gUFMdV"
      >
        <div
          class="pagination"
        >
          <p
            data-testid="pageNumberCopy"
          >
            Page 1 of 4
          </p>
          <div
            class="buttons"
          >
            <button
              class="btn"
              data-testid="previousPage"
              disabled=""
              type="button"
            >
              Previous
            </button>
            <button
              class="btn"
              data-testid="nextPage"
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
